src/Cake.Common/Tools/VSWhere/Product/VSWhereProduct.cs

```diff
@@ -55,6 +56,10 @@ private ProcessArgumentBuilder GetArguments(VSWhereProductSettings settings)
             {
                 builder.AppendQuoted(settings.Products);
             }
+            else
+            {
+                builder.AppendQuoted("*");
```

Comment:
Calling without * causes a fail for we with exit code 87 (Error 0x57: Argument required for: products)